### PR TITLE
Plugin refresh and cd

### PR DESCRIPTION
**Pull request by jtnord**
Update the plugin baseline and enable CD

depends on https://github.com/jenkins-infra/repository-permissions-updater/pull/2722

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by rsandell — APPROVED** [review]
🏇

File: src/main/resources/index.jelly [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<description>
+    This plugin provides support for iterating through all the <a href="https://javadoc.jenkins.io/jenkins/model/Nodes.html">Node</a>
+    instances that are in use by <code>Jenkins</code>, even those <code>Node</code> instances that are not traditionally attached to Jenkins. The API exposed by this
+    plugin can be used by cloud provider plugins to identify unused provisioned resource.
+  </description>
```
Comment (basil) [inline src/main/resources/index.jelly:27]: Inconsistent indentation:

```suggestion
</description>
```

Also, per https://github.com/jenkinsci/jenkins/blob/44d0496f38584dcc9fffc42de4104b66753649a3/core/src/main/resources/hudson/PluginManager/installed.jelly#L110-L129 the description from `src/main/resources/index.jelly` is preferred over the `<description>` from `pom.xml` ("last resort" according to the Jenkins core source code), making the `<description>` from `pom.xml` now redundant. That is why https://github.com/jenkinsci/maven-hpi-plugin/blob/694a97af11fdc7b841ee220c17ed64dfaffe08b0/src/main/java/org/jenkinsci/maven/plugins/hpi/HpiMojo.java#L139-L143 says:

> Delete any `<description>` from `pom.xml` and create `src/main/resources/index.jelly`

This PR creates `src/main/resources/index.jelly`, but it does not delete the `<description>` from `pom.xml`.

**Review by basil — APPROVED** [review]

File: src/main/resources/index.jelly (hunk 1 above)
Comment (jtnord) [reply src/main/resources/index.jelly:27]: Description is a maven thing that has uses outside Jenkins, so I explicitly ignored that section.

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -1,3 +1 @@
-_extends: .github
-tag-template: node-iterator-api-$NEXT_MINOR_VERSION
-name-template: Node Iterator API Plugin $NEXT_MINOR_VERSION
+.github/release-drafter.yml
```
[the remark this answers is not in the document]
Comment (jtnord) [reply .github/release-drafter.yml:1]: Good spot. Looks like my copy paste buffer was not update.

File: src/main/resources/index.jelly (hunk 1 above)
Comment (basil) [reply src/main/resources/index.jelly:27]: It does have theoretical uses outside of Jenkins, but I don't think there are any such uses in practice, and I think having the same information copied and pasted in multiple places slightly increases maintenance burden, so for that reason and consistency with existing precedent I still have a slight preference for deleting the `<description>` from `pom.xml`, but I don't feel strongly about this.

I do still think that the indentation should be made consistent by deleting the spaces, at the beginning of the line, but I wouldn't block this PR over some whitespace. I already approved the PR.

File: .github/release-drafter.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jtnord) [reply .github/release-drafter.yml]: fixed in  03e0f31

File: src/main/resources/index.jelly (hunk 1 above)
Comment (jtnord) [reply src/main/resources/index.jelly]: the invalid html has been fixed `description` -> `div` along with the indent in 03e0f31

File: .github/dependabot.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name: cd
```
Comment (jtnord) [inline .github/dependabot.yaml:1]: this is not a dependabot file... #7

**Pull request merged**